ledger/stateproofverificationtracker.go

```diff
@@ -165,10 +172,24 @@ func (spt *stateProofVerificationTracker) close() {
 }
 
 func (spt *stateProofVerificationTracker) LookupVerificationData(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationData, error) {
+	verificationData, err := spt.lookUpVerificationData(stateProofLastAttestedRound)
```

Comment:
In this design, when there is a cache hit, we will update the cache where it is not needed.

maybe we can just look at the cache first and then take the regular flow?